src/server/v2.0/handler/auditlog.go

```diff
@@ -21,14 +26,34 @@ type auditlogAPI struct {
 }
 
 func (a *auditlogAPI) ListAuditLogs(ctx context.Context, params auditlog.ListAuditLogsParams) middleware.Responder {
-	// ToDo enable permission check
-	// if !a.HasPermission(ctx, rbac.ActionList, rbac.ResourceLog) {
-	//	return a.SendError(ctx, ierror.ForbiddenError(nil))
-	// }
+	secCtx, ok := security.FromContext(ctx)
+	if !ok {
+		return a.SendError(ctx, ierror.UnauthorizedError(errors.New("security context not found")))
+	}
+	if !secCtx.IsAuthenticated() {
+		return a.SendError(ctx, ierror.UnauthorizedError(nil).WithMessage(secCtx.GetUsername()))
+	}
 	query, err := a.BuildQuery(ctx, params.Q, params.Page, params.PageSize)
 	if err != nil {
 		return a.SendError(ctx, err)
 	}
+
+	if !secCtx.IsSysAdmin() {
+		// ToDo remove the dependency on GetMyProjects()
+		projects, err := secCtx.GetMyProjects()
+		if err != nil {
+			return a.SendError(ctx, fmt.Errorf(
+				"failed to get projects of user %s: %v", secCtx.GetUsername(), err))
+		}
+		ids := []int64{}
+		for _, project := range projects {
+			if a.HasProjectPermission(ctx, project.ProjectID, rbac.ActionList, rbac.ResourceLog) {
+				ids = append(ids, project.ProjectID)
+			}
+		}
+		query.Keywords["ProjectIDs"] = ids
```

Comment:
Let's use `ProjectID` instead. The `ProjectIDs` isn't a property of `Log` model. The API consumers know nothing about it if we don't document it